rust/src/memory.rs

```diff
@@ -79,6 +79,13 @@ extern "C" {
     pub fn memcmp(p1: *const u8, p2: *const u8, len: usize) -> i32;
 }
 
+/// Check if the pointer `p` is aligned to offset `a`.
+pub fn is_aligned<T>(p: *const T, a: usize) -> bool {
```

Comment:
I know this is from libcore, but could we have some rough tests for it?

Reply:
Done.